nepomuk/src/main/java/edu/asu/diging/gilesecosystem/nepomuk/core/service/handlers/AbstractFileHandler.java

```diff
@@ -104,5 +104,15 @@ public String getRelativePathInTypeFolder(IFile file) {
     
     protected abstract IFileStorageManager getStorageManager();
     
+    public void deleteFile(IFile file, boolean isOldFileVersion) {
+        if(!isOldFileVersion) {
+            try {
+                getStorageManager().deleteFile(file.getUsername(), file.getUploadId(), file.getDocumentId(), file.getFilename()); 
+            } catch (NepomukFileStorageException e) {
+                messageHandler.handleMessage("File could not be deleted.", e, MessageType.WARNING);
```

Comment:
if this happens, the fileManager should not be called. Otherwise, you get inconsistent data.

Reply:
It seems like both, the catch and the `isOldFileVersion` parameter try to achieve the same thing, if there is no file to delete, just delete the db entry. Instead of this, just check if the file exists, if it does, delete it, if not just move on to deleting the db entry. Is this the only reason `isOldFileVersion` exists? If so, this can be deleted as well.